### PR TITLE
caf: power_manager: Postpone system off state enter

### PR DESCRIPTION
**Pull request by rakons**
This commit adds a delay between receiving the power off request and powering the CPU down.
This allows processing of wake up events that are in the event queue, generated by some modules in reaction on power down event.

JIRA: NCSDK-24656

### CODE REVIEW

File: subsys/caf/modules/power_manager.c [hunk 1]
```diff
@@ -153,6 +151,18 @@ static void system_off_on_error(void)
 	sys_poweroff();
 }
 
+static void system_off_handler(struct k_work *work)
+{
+	LOG_WRN("System turned off");
+	LOG_PANIC();
```
Comment (gmarull) [inline subsys/caf/modules/power_manager.c:157]: won't panic halt?

Comment (rakons) [reply subsys/caf/modules/power_manager.c:157]: LOG_PANIC means only to process all logs.

**Review by gmarull — APPROVED** [review]

**Review by zycz — APPROVED** [review]
Ok, then it's fine.

**Comment by gte1** [comment]
Could this be added to milestone 2.5.1 ?

**Pull request merged**